asyncthreadpool.nim

```diff
@@ -1,6 +1,11 @@
-import macros, cpuinfo, asyncdispatch
+import macros, cpuinfo
 import ./private/pipes
 
+when not defined(ChronosAsync):
```

Comment:
I wonder if `-d:ChronosAsync` is some existing convention for such cases? Maybe it should be documented in chronos.

Reply:
it used in my `amysql` lib, for support both std and chronos asyncdispatch, dont know if it needed to be defined by chronos.

Reply:
Just a thought. Supporting both chronos and asyncdispatch is a use case common enough to be done uniformly. E.g. there's another [news](https://github.com/tormund/news) library that also supports both asyncdispatch and chronos but does it differently. So it is in chonos' authors interests to provide some common ground for such use cases, so in the end chronos users benefit from it.

Reply:
Asked this question on chronos: https://github.com/status-im/nim-chronos/issues/207